### PR TITLE
Add update function

### PR DESCRIPTION
**Pull request by dentrado**
Adds an `update` function in addition to the existing `updateIn`, similar to how there's both `set` and `setIn`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -270,6 +270,27 @@ setIn({ a: 3 }, ['unknown', 0, 'path'], 4)
 // { a: 3, unknown: [{ path: 4 }] }
 ```
 
+#### update()
+Returns a new object with a modified attribute,
+calculated via a user-provided callback based on the current value.
+If the calculated value is the same (*referentially equal to*)
+the previous value, the original object is returned.
```
Comment (lizell) [inline README.md]: Missing an "as" after the parenthesis?

File: src/timm.js [hunk 2]
```diff
@@ -361,6 +361,36 @@ export function setIn<T: ArrayOrObject>(obj: T, path: Array<Key>, val: any): T {
   return doSetIn(obj, path, val, 0);
 }
 
+// -- #### update()
+// -- Returns a new object with a modified attribute,
+// -- calculated via a user-provided callback based on the current value.
+// -- If the calculated value is the same (*referentially equal to*)
```
Comment (lizell) [inline src/timm.js]: Same here

**Comment by guigrpa** [comment]
Thanks for the interesting PR!

**Comment by guigrpa** [comment]
Published in v1.3.0

**Pull request merged**